src/lib.rs

```diff
@@ -16,6 +16,28 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with Flue. If not, see <https://www.gnu.org/licenses/>.
 
+//! Flue is an efficient and secure actor runtime library.
+//!
+//! The fundamental building block in Flue is the **process**: a concurrent
+//! execution thread with private memory. Processes may only communicate by
+//! passing **signals** between each other. Signals are sent to **routes** and
+//! received using **mailboxes**. **Capabilities** both reference and limit
+//! access to routes using fine-grained permission flags. **Tables** are stores
+//! of integer-addressed, unforgeable capabilities.
+//!
+//! Flue is made for the purpose of efficiently executing potentially untrusted
+//! process code, so to support running that code, Flue's security model has
+//! the following axioms:
+//! - Routes can only be accessed by a process if a capability to that route
+//!   has explicitly been passed to that process in a signal. You may sandbox
+//!   processes on a fine-grained level by limiting which capabilities get
+//!   passed to it.
+//! - Capabilities may only be created from other capabilities with either an
+//!   identical set or a subset of the original's permissions. Permission
+//!   escalation by untrusted processes is impossible.
+
+#![warn(missing_docs)]
+
```

Comment:
What would be useful forna technical library like this is an explanation of the lifetime of a message, where it starts and the route it takes. This would give a great overview of how all the structs relate and what their purposw in the act of sending and receiving a message is ❤️